### PR TITLE
feat: update ECS config and apply standards

### PR DESCRIPTION
**Pull request by nlemoine**
Bump ECS config.

Note it changed quite a lot a files and we might want to add this commit in the .git-blame-ignore-revs file.

It also removed to full namespace from the docblocks, I don't if this cause issues when generating API reference? @gchtr

### CODE REVIEW

**Comment by Levdbas** [comment]
I see we are no longer prefixing the docblock return types with the full namespace. Could this have a negative impact in the symbol discovery of IDE's?

**Comment by nlemoine** [comment]
No, since they're referenced in imports: https://github.com/timber/timber/blob/77bdbdb0640c3057fd6489f43d08b266e39e474c/src/Factory/PagesMenuFactory.php#L5

I just wonder if the doc generation tool is able to compute the FQCN or is it parsing the doc block as is, example:
Capture d’écran 2024-01-26 à 12 34 09

**Comment by Levdbas** [comment]
I've just tested the doc generation in combination of your branch and the docs still seem to compile the correct paths :)

File: src/MenuItem.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ class MenuItem extends CoreEntity
      *
      * @since 1.12.0
      * @see \Timber\MenuItem::menu()
-     * @var \Timber\Menu The `Timber\Menu` object the menu item is associated with.
+     * @var Menu The `Timber\Menu` object the menu item is associated with.
```
Comment (Levdbas) [inline src/MenuItem.php]:
```suggestion
     * @var Menu The `Menu` object the menu item is associated with.
```

File: src/MenuItem.php [hunk 2]
```diff
@@ -119,8 +119,8 @@ class MenuItem extends CoreEntity
     /**
      * @internal
      * @param array|object $data The data this MenuItem is wrapping
-     * @param \Timber\Menu $menu The `Timber\Menu` object the menu item is associated with.
-     * @return \Timber\MenuItem a new MenuItem instance
+     * @param Menu $menu The `Timber\Menu` object the menu item is associated with.
```
Comment (Levdbas) [inline src/MenuItem.php]:
```suggestion
     * @param Menu $menu The `Menu` object the menu item is associated with.
```

File: src/MenuItem.php [hunk 3]
```diff
@@ -130,7 +130,7 @@ public static function build($data, ?Menu $menu = null): self
     /**
      * @internal
      * @param WP_Post $data
-     * @param \Timber\Menu $menu The `Timber\Menu` object the menu item is associated with.
+     * @param Menu $menu The `Timber\Menu` object the menu item is associated with.
```
Comment (Levdbas) [inline src/MenuItem.php]:
```suggestion
     * @param Menu $menu The `Menu` object the menu item is associated with.
```

File: src/MenuItem.php [hunk 4]
```diff
@@ -479,7 +479,7 @@ public function target()
      *
      * @api
      * @since 1.12.0
-     * @return \Timber\Menu The `Timber\Menu` object the menu item is associated with.
+     * @return Menu The `Timber\Menu` object the menu item is associated with.
```
Comment (Levdbas) [inline src/MenuItem.php]:
```suggestion
     * @return Menu The `Menu` object the menu item is associated with.
```

File: src/AccessesPostsLazily.php [hunk 5]
```diff
@@ -67,7 +66,7 @@ trait AccessesPostsLazily
      *   // No additional overhead here.
      * }
      * ```
-     * @return \Timber\PostCollectionInterface The realized PostQuery.
+     * @return PostCollectionInterface The realized PostQuery.
```
Comment (Levdbas) [inline src/AccessesPostsLazily.php:69]: This throws a static analysis error. Is this something we must take care of? Not sure if this was something that we already discussed in an issue.

**Review by Levdbas — COMMENTED** [review]
I suggested some docblock updates while at it and there is one return type issue that we might already discussed in a ticket. Just checking. Other than that, looks good to me.

File: src/AccessesPostsLazily.php (hunk 5 above)
Comment (Levdbas) [reply src/AccessesPostsLazily.php:69]: @gchtr , mentioned this over here: https://github.com/timber/timber/pull/2668

Comment (nlemoine) [reply src/AccessesPostsLazily.php:69]: Haven't figured it out yet but this shouldn't prevent us from merging the PR. We don't require PHPStan to pass (yet). :)

**Comment by Levdbas** [comment]
@gchtr , can you take a final look at this?

**Comment by gchtr** [comment]
> It also removed to full namespace from the docblocks, I don't if this cause issues when generating API reference? @gchtr

It’s not a problem, because the reflections still work and generate fully qualified names, apparently.

It only **doesn’t work** for hook documentation. There, the types have to be generated from the DocBlock. As you can see in the following screenshot, we could have `Post` or `Environment`, but wouldn’t know whether this comes from `Timber` or `Twig`.



These hook DocBlocks are parsed by looping over the tokens in https://github.com/timber/teak/blob/a5af1504cadb8e5d2277069165caf948d1d0c012/lib/Compiler/HookReference.php#L62-L70. I guess I could somehow also derive the types from the variables that are present before a hook appears. I’ll have to check this out. I created an issue for this in https://github.com/timber/teak/issues/3.

For now, I think it’s a tradeoff we can live with, because it only affects some few filters in https://timber.github.io/docs/v2/hooks/filters/.

**Review by gchtr — APPROVED** [review]

**Comment by nlemoine** [comment]
Thanks @gchtr for the details about doc generation. There should be a way to grab the FQCN, let me know if I can help.

**Pull request merged**